src/h.js

```diff
@@ -1,22 +1,18 @@
 export default function(tag, data) {
-  var node
-  var stack = []
   var children = []
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
-  }
-
-  while (stack.length) {
-    if (Array.isArray((node = stack.pop()))) {
-      for (var i = node.length; i--; ) {
-        stack[stack.length] = node[i]
+  for (var i = 2; i < arguments.length; i++) {
+    var val = arguments[i]
+    if (Array.isArray(val)) {
+      for (var j = 0; j < val.length; j++) {
+        children.push(val[j])
       }
-    } else if (node != null && node !== true && node !== false) {
-      if (typeof node === "number") {
-        node = node + ""
+    }
+    else if (val && val !== true) {
```

Comment:
@ngryman You want:

```js
} else if (val != null && val !== true && val !== false) {
```

Otherwise we will not be able to process text nodes with the number 0.